src/scripts/main.js

```diff
@@ -1,3 +1,264 @@
+    </label>
+    <button
+      id="save_button" name="button" type="text" data-qa="salary"
+    >
+      Save to table
+    </button>
+  </for>
+`);
+
+function pushNotification(posTop, posRight, title, description, type) {
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('afterend', `
+    <div class="notification ${type}" data-qa="notification">
+      <h2 class="title">${title}</h2>
+      <p>${description}</p>
+    </div>
+  `);
+
+  const notification = document.querySelector('div');
+
+  notification.style.top = posTop + 'px';
+  notification.style.right = posRight + 'px';
+
+  setTimeout(() => notification.remove(), 5000);
+};
+
+function checkForErrors(employeeName, position, office, age, salary) {
+  const errorMessages = [];
+
+  if (employeeName.length > 0 && employeeName.length < 4) {
+    errorMessages.push('Name should consist of at least 4 letters.');
+  }
+
+  if (age !== '' && (+age < 18 || +age > 90)) {
+    errorMessages.push('Age range should be from 18 to 90.');
+  }
+
+  if (
+    employeeName === ''
+    || position === ''
+    || office === ''
+    || age === ''
+    || salary === ''
+  ) {
+    errorMessages.push('All fields are required.');
+  }
+
+  if (salary < 0) {
+    errorMessages.push('Salary should be larger than 0.');
+  }
+
+  if (Number.isNaN(salary)) {
+    errorMessages.push('Salary should be numeric value.');
+  }
+
+  return errorMessages;
+}
+
+const form = document.forms[0];
+
+form.addEventListener('submit', (handler) => {
+  handler.preventDefault();
+
+  const data = new FormData(form);
+  const salary = data.get('salary');
+  const employeeName = data.get('name');
+  const position = data.get('position');
+  const office = data.get('office');
+  const age = data.get('age');
+  const salaryString = `$` + Number(salary).toLocaleString();
+  const errorMessages = checkForErrors(
+    employeeName, position, office, age, salary
+  );
+
+  if (errorMessages.length) {
+    pushNotification(
+      10, 10, 'There was an error:', errorMessages.join('\n'), 'error'
+    );
+  } else {
+    pushNotification(
+      10, 10, 'Success', 'Employee was saved successfully.', 'success'
+    );
+
+    tableBody.insertAdjacentHTML('beforeend', `
+    <tr>
+      <td>${employeeName}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${salaryString}</td>
+    </tr>
+    `);
+
+    form.reset();
+  }
+});
+
+tableBody.addEventListener('dblclick', (e) => {
+  const cell = e.target;
+  const cellIndex = cell.cellIndex;
+  const initialValue = cell.innerText;
+  const selectedRow = e.target.closest('tr');
+  const input = document.createElement('input');
+
+  input.classList.add('cell-input');
+  input.type = 'text';
+  input.name = 'cell-input';
+  input.setAttribute('value', initialValue);
+  cell.innerText = '';
+  cell.append(input);
+  input.focus();
+
+  input.addEventListener('keyup', (f) => {
```

Comment:
I know that lint doesn't allow you repeat variable names in one file, but try to rename it (`f`) somehow.

Reply:
Changed it to (handleKeypress).